api/src/main/java/org/eclipse/microprofile/faulttolerance/Fallback.java

```diff
@@ -40,8 +49,19 @@
 
     /**
      * Specify the fallback class to be used. An new instance of the fallback class
-     * is returned. The instance is unmanaged.
+     * is returned. The instance is unmanaged. The type parameter of the fallback class must be assignable to the
+     * return type of the annotated method. Otherwise, the {@link javax.enterprise.inject.spi.DeploymentException} must be thrown.
+     * 
      * @return the fallback class
      */
     Class<? extends FallbackHandler<?>> value();
+    /**
+    * Specify the method name to be fallbacked to. This method belongs
+    * to the same class as the method to fallback.
+    * The method must be either zero argument or have the exactly same arguments as the method being annotated.
```

Comment:
Why would zero argument be useful?  Should we also support `ExecutionContext` as a parameter?